src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -356,10 +357,46 @@ public InsertValidationResponse insertRows(
                 this.statsMap.put(
                     colName,
                     RowBufferStats.getCombinedStats(stats, this.tempStatsMap.get(colName))));
-      }
+        this.channelState.setOffsetToken(offsetToken);
+      } else if (onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
```

Comment:
For readability and to reduce the size of the function should we pull the below into its own function?